docs/source/user-guide/supported_models.rst

```diff
@@ -0,0 +1,56 @@
+.. _supported_models:
+
+====================
+Supported Models
+====================
+
+Deepchecks supports a minmal version of the sklearn api.
+
+
+Required API
+=============
+
+Regression
+----------
+
+All that is necessary for for regression models is the ``predict`` function.
+
+Example of an simple model:
+
+>>> class simple_regression_model:
+...     def predict(X: pd.DataFrame) -> pd.Series:
+...         ...
+
+Classification
+--------------
+
+For classification models, we require also the ``predict_proba`` function:
+
+>>> class simple_classification_model:
+...     def predict(X: pd.DataFrame) -> pd.Series:
```

Comment:
Maybe we should include the return format?

Reply:
indeed

want to note that these examples are a bit confusing as it seems that they should all receive a dataframe and return a series, while this isn't necessarily the (only) standard